test/gtest/ucm/malloc_hook.cc

```diff
@@ -355,28 +356,61 @@ UCS_TEST_F(malloc_hook, fork) {
 class malloc_hook_cplusplus : public malloc_hook {
 public:
 
-    malloc_hook_cplusplus() : m_unmapped_size(0) {
+    malloc_hook_cplusplus() : m_mapped_size(0), m_unmapped_size(0) {
     }
 
-    static void mem_event_callback(ucm_event_type_t event_type, ucm_event_t *event,
-                                   void *arg)
-    {
-        malloc_hook_cplusplus *self = reinterpret_cast<malloc_hook_cplusplus*>(arg);
-        self->m_unmapped_size += event->vm_unmapped.size;
+    void set() {
+        ucs_status_t status;
+        status = ucm_set_event_handler(UCM_EVENT_VM_MAPPED|UCM_EVENT_VM_UNMAPPED,
+                                       0, mem_event_callback,
+                                       reinterpret_cast<void*>(this));
+        ASSERT_UCS_OK(status);
+    }
+
+    void unset() {
+        ucm_unset_event_handler(UCM_EVENT_VM_MAPPED|UCM_EVENT_VM_UNMAPPED,
+                                mem_event_callback, reinterpret_cast<void*>(this));
     }
 
 protected:
+    static void mem_event_callback(ucm_event_type_t event_type,
+                                   ucm_event_t *event, void *arg)
+    {
+        malloc_hook_cplusplus *self =
+                        reinterpret_cast<malloc_hook_cplusplus*>(arg);
+        switch (event_type) {
+        case UCM_EVENT_VM_MAPPED:
+            self->m_mapped_size   += event->vm_mapped.size;
+            break;
+        case UCM_EVENT_VM_UNMAPPED:
+            self->m_unmapped_size += event->vm_unmapped.size;
+            break;
+        default:
+            break;
+        }
+    }
+
+    double measure_alloc_time(size_t size, unsigned iters)
+    {
+        ucs_time_t start_time = ucs_get_time();
+        for (unsigned i = 0; i < iters; ++i) {
+            void *ptr = malloc(size);
+            *(char*)ptr = '5';
```

Comment:
you just touch the memory and 5 has no any special meaning, right?

Reply:
yes, i want to prevent the compiler from removing that code

Reply:
add comment, otherwise somebody else will remove it